src/lib/transition_frontier/catchup_hash_tree.ml

```diff
@@ -15,14 +25,45 @@ module Node = struct
           (* If a node has a breadcrumb, then all of its ancestors have
    breadcrumbs as well. *)
     }
+  [@@deriving to_yojson]
+end
+
+module L = struct
+  type t = Logger.t
+
+  let to_yojson _ = `String "logger"
+end
+
+module State_hash_table = struct
+  type 'a t = 'a State_hash.Table.t
+
+  let to_yojson f t : Yojson.Safe.t =
+    `Assoc
+      (List.map (State_hash.Table.to_alist t) ~f:(fun (h, x) ->
+           (State_hash.to_base58_check h, f x) ))
+end
+
+module State_hash_hash_set = struct
+  type t = State_hash.Hash_set.t
+
+  let to_yojson (t : t) : Yojson.Safe.t =
+    `List (List.map (Hash_set.to_list t) ~f:State_hash.to_yojson)
+end
+
+module State_hash_set = struct
+  type t = State_hash.Set.t
+
+  let to_yojson (t : t) : Yojson.Safe.t =
+    `List (List.map (Set.to_list t) ~f:State_hash.to_yojson)
 end
 
 type t =
-  { nodes: Node.t State_hash.Table.t
-  ; tips: State_hash.Hash_set.t
-  ; children: State_hash.Set.t State_hash.Table.t
+  { nodes: Node.t State_hash_table.t
+  ; tips: State_hash_hash_set.t
+  ; children: State_hash_set.t State_hash_table.t
   ; mutable root: State_hash.t
-  ; logger: Logger.t }
+  ; logger: L.t }
```

Comment:
you could use `Logger.t` as the type here, and add an annotation: `[@to_yojson fun _ -> `String "logger"]`

instead of creating the module `L`